xcell/mappers/mapper_eBOSSQSO.py

```diff
@@ -21,82 +21,96 @@ def __init__(self, config):
         """
         self._get_defaults(config)
 
-        self.cat_data = []
-        self.cat_random = []
-        self.z_arr_dim = config.get('z_arr_dim', 50)
+        self.cats = {'data': [], 'random': []}
 
-        for file_data, file_random in zip(self.config['data_catalogs'],
-                                          self.config['random_catalogs']):
-            if not os.path.isfile(file_data):
-                raise ValueError(f"File {file_data} not found")
-            with fits.open(file_data) as f:
-                self.cat_data.append(Table.read(f))
-            if not os.path.isfile(file_random):
-                raise ValueError(f"File {file_random} not found")
-            with fits.open(file_random) as f:
-                self.cat_random.append(Table.read(f))
-
-        self.cat_data = vstack(self.cat_data)
-        self.cat_random = vstack(self.cat_random)
+        self.z_arr_dim = config.get('z_arr_dim', 50)
         self.nside_mask = config.get('nside_mask', 512)
         self.npix = hp.nside2npix(self.nside)
-
         self.z_edges = config['z_edges']
 
-        self.cat_data = self._bin_z(self.cat_data)
-        self.cat_random = self._bin_z(self.cat_random)
-        self.w_data = self._get_weights(self.cat_data)
-        self.w_random = self._get_weights(self.cat_random)
-        self.alpha = np.sum(self.w_data)/np.sum(self.w_random)
+        self.ws = {'data': None, 'random': None}
+        self.alpha = None
 
         self.dndz = None
         self.delta_map = None
         self.nl_coupled = None
         self.mask = None
 
+    def get_catalog(self, mod='data'):
+        if mod == 'data':
+            data_file = self.config['data_catalogs']
+        else:
+            data_file = self.config['random_catalogs']
+
+        if len(self.cats[mod]) == 0:
```

Comment:
And as before:
```
if self.cats[mod] == None:
    self.cats[mod] = []
```

Reply:
I see! I was struggling with that. Thanks!

Reply:
resolved!